packages/devtools_app/lib/src/timeline/timeline_controller.dart

```diff
@@ -248,21 +249,11 @@ class TimelineController {
 
 class FrameBasedTimeline
     extends TimelineBase<FrameBasedTimelineData, FrameBasedTimelineProcessor> {
-  /// Stream controller that notifies a frame was added to the timeline.
-  ///
-  /// Subscribers to this stream will be responsible for updating the UI for the
-  /// new value of [frameBasedTimelineData.frames].
-  final _frameAddedController = StreamController<TimelineFrame>.broadcast();
+  /// Notifies that a frame has been added to the timeline.
+  final frameAddedNotifier = ValueNotifier<TimelineFrame>(null);
```

Comment:
An idea for discussion: If we often need to listen to all of these notifiers, then we could alternately make this class implement `Listenable`.  Listening to the controllers would then involve registering a single listener that we would notify when any field on the class changed.